crates/latch_core/src/ecs/storage/column.rs

```diff
@@ -48,7 +48,7 @@ impl Column {
                 Vec::with_capacity(initial_bytes),
                 Vec::with_capacity(initial_bytes),
             ],
-            capacity: Self::INITIAL_CAPACITY,
+            capacity: 0,  // Start with 0 capacity, will grow on first alloc
```

Comment:
We lazily initialize columns only when we need to store data in them, so there will ALWAYS be an initial allocation shortly after this. Delaying the capacity allocation just means adding an unnecessary conditional statement to our capacity check. Please revert this.